Extension/src/LanguageServer/extension.ts

```diff
@@ -184,7 +186,9 @@ export async function activate(): Promise<void> {
 
     disposables.push(vscode.workspace.onDidChangeConfiguration(onDidChangeSettings));
     disposables.push(vscode.window.onDidChangeActiveTextEditor(onDidChangeActiveTextEditor));
-    ui.activeDocumentChanged(); // Handle already active documents (for non-cpp files that we don't register didOpen).
+    if (ui instanceof OldUI) {
+        void ui?.activeDocumentChanged(); // Handle already active documents (for non-cpp files that we don't register didOpen).
```

Comment:
Why is the activeDocumentChanged code not run for the newUI? The intent of that code was to enable the configuration switching when c_cpp_properties.json was made active.

Reply:
Actually, there's a regression in that this code in activeDocumentChanged is no longer being hit:
```ts
            const isCpp: boolean = (activeEditor.document.uri.scheme === "file" && (activeEditor.document.languageId === "c" || activeEditor.document.languageId === "cpp" || activeEditor.document.languageId === "cuda-cpp"));

            let isCppPropertiesJson: boolean = false;
            if (activeEditor.document.languageId === "json" || activeEditor.document.languageId === "jsonc") {
                isCppPropertiesJson = activeEditor.document.fileName.endsWith("c_cpp_properties.json");
                if (isCppPropertiesJson) {
                    vscode.languages.setTextDocumentLanguage(activeEditor.document, "jsonc");
                }
            }
```
i.e. the c_cpp_properties.json should get associated with jsonc instead of json.

Reply:
In the new UI that is being handled by the documentSelector. Configuration has its own filter that includes json and jsonc files
https://github.com/microsoft/vscode-cpptools/blob/b522b62515a6117f3510d90d17bf3a582bfdefaf/Extension/src/LanguageServer/ui_new.ts#L52-59
```ts
    private configDocumentSelector: vscode.DocumentFilter[] = [
        { scheme: 'file', language: 'c' },
        { scheme: 'file', language: 'cpp' },
        { scheme: 'file', language: 'cuda-cpp' },
        { scheme: 'file', language: 'jsonc', pattern: '**/.vscode/*.json'},
        { scheme: 'file', language: 'jsonc', pattern: '**/*.code-workspace'},
        { scheme: 'output'}
    ];
```

Reply:
@Bernardin-MS No, the code that associates c_cpp_properties.json is not being handled by the document selector, i.e. you need to keep that functionality in activeDocumentChanged, otherwise it's a regression bug and c_cpp_properties.json will get associated with json instead of jsonc, so users with existing comments will start getting warnings.